output-gazetteer.cpp

```diff
@@ -692,22 +692,20 @@ int output_gazetteer_t::process_way(osmid_t id, osmid_t *ndv, int ndc,
 
     /* Are we interested in this item? */
     if (places.has_data()) {
-        struct osmNode *nodev;
         int nodec;
 
+        assert(ndc > 0 && "no nodes to proccess");
```

Comment:
The current code actually produces empty geometries for 0-node ways. Simply ignoring them might leave zombies when running in append mode. Better to dump empty ways at the very beginning of the function as follows:
~~~
  if (ndc <= 0) {
    delete_unused_full('W', id);
    return 0;
  }
~~~